### PR TITLE
CCD-3659 Adjust GlobalSearch CaseType validation and auto-population

### PR DESCRIPTION
**Pull request by mattnayler**
### JIRA link (if applicable) ###

   [CCD-3659](https://tools.hmcts.net/jira/browse/CCD-3659) _"If IAC search for a civil case using GS then no results are shown."_

### Change description ###

* Adjust GlobalSearch request criteria validation: at least one jurisdiction or case type must be provided in the search criteria.
* Adjust GlobalSearch case type auto population that is used when search value is null or empty.  It will now populate from Jurisdiction lookup.

> NB: ES test PRs are:
> * hmcts/ccd-data-store-api#2093
> * hmcts/ccd-definition-store-api#1283
>
> **Both PRs above should be closed once this PR is merged**

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes :: validation on GlobalSearch API changed to  make one of CaseTypes or Jurisdictions mandatory.
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/model/std/validator/globalsearch/SearchCriteriaValidator.java [hunk 1]
```diff
@@ -1,24 +1,19 @@
...
     public boolean isValid(final SearchCriteria criteria, final ConstraintValidatorContext context) {
         if (criteria != null) {
-            try {
-                return criteria.getNonNullFields();
-            } catch (IllegalAccessException e) {
-                LOG.error(e.getMessage());
-            }
+            // rule: At least one jurisdiction or case type must be provided in the search criteria
+            return !(CollectionUtils.isEmpty(criteria.getCcdJurisdictionIds()))
+                || !(CollectionUtils.isEmpty(criteria.getCcdCaseTypeIds()));
```
Comment (danlysiak) [inline src/main/java/uk/gov/hmcts/ccd/domain/model/std/validator/globalsearch/SearchCriteriaValidator.java:16]:
```suggestion
            return !CollectionUtils.isEmpty(criteria.getCcdJurisdictionIds())
                || !CollectionUtils.isEmpty(criteria.getCcdCaseTypeIds());
```
Minor suggestion to remove unnecessary extra brackets.

**Review by danlysiak — APPROVED** [review]
Approved implementation; one not very important comment!

**Review by marianadpereira — APPROVED** [review]
The tests look good to me.

File: src/aat/resources/features/F-1005 - GlobalSearch - Search cases/S-1005.11.td.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+  "title": "Unsuccessful search for case with neither CaseTypes or Jurisdictions in search criteria",
+
+  "_guid_": "S-1005.11",
+  "_extends_": "F-1005_SearchCaseBase",
+
+  "specs": [
+    "an active profile in CCD",
+
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/aat/resources/features/F-1005 - GlobalSearch - Search cases/S-1005.11.td.json]: Resolved in 59de50972eebafcca91d7c5a4c1b52c6e3d1f4f6.

[the remark this answers is not in the document]
Comment (mattnayler) [reply src/aat/resources/features/F-1005 - GlobalSearch - Search cases/S-1005.11.td.json]: Resolved in 59de50972eebafcca91d7c5a4c1b52c6e3d1f4f6.

File: src/aat/resources/features/F-1005 - GlobalSearch - Search cases/S-1005.12.td.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
+{
+  "title": "Successfully search for case with CaseTypes but not Jurisdictions in search criteria",
+
+  "_guid_": "S-1005.12",
+  "_extends_": "F-1005_SearchCaseBase",
+
+  "specs": [
+    "an active profile in CCD",
+
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/aat/resources/features/F-1005 - GlobalSearch - Search cases/S-1005.12.td.json]: Resolved in 59de50972eebafcca91d7c5a4c1b52c6e3d1f4f6.

**Pull request merged**